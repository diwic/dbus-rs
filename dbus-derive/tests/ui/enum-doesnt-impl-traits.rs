use dbus_macros::DbusEnum;

#[derive(DbusEnum)]
#[dbus_enum(as_type = "u8")]
pub enum EnumArg {
    Opt0,
    Opt1,
}

fn main() {}
