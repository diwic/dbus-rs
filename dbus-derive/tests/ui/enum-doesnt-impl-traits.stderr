error[E0277]: the trait bound `u8: From<EnumArg>` is not satisfied
 --> tests/ui/enum-doesnt-impl-traits.rs:4:23
  |
4 |   #[dbus_enum(as_type = "u8")]
  |  _______________________^
5 | | pub enum EnumArg {
  | |________________^ the trait `From<EnumArg>` is not implemented for `u8`
  |
  = help: the following other types implement trait `From<T>`:
            <u8 as From<bool>>
            <u8 as From<NonZeroU8>>
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `EnumArg: From<u8>` is not satisfied
 --> tests/ui/enum-doesnt-impl-traits.rs:4:23
  |
4 |   #[dbus_enum(as_type = "u8")]
  |  _______________________^
5 | | pub enum EnumArg {
  | |________________^ the trait `From<u8>` is not implemented for `EnumArg`
  |
  = note: required for `u8` to implement `Into<EnumArg>`
  = note: required for `EnumArg` to implement `TryFrom<u8>`
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
